src/sentry/static/sentry/app/components/group/sidebar.tsx

```diff
@@ -53,44 +54,55 @@ class GroupSidebar extends React.Component<Props, State> {
   }
 
   componentDidMount() {
+    this.fetchAllEnvironmentsGroupData();
+    this.fetchCurrentRelease();
+    this.fetchParticipants();
+    this.fetchTagData();
+  }
+
+  UNSAFE_componentWillReceiveProps(nextProps) {
+    if (!isEqual(nextProps.environments, this.props.environments)) {
+      this.setState({environments: nextProps.environments}, this.fetchTagData);
+    }
+  }
+
+  fetchAllEnvironmentsGroupData() {
     const {group, api} = this.props;
 
-    api.request(`/issues/${group.id}/participants/`, {
+    // Fetch group data for all environments since the one passed in props is filtered for the selected environment
+    // The charts rely on having all environment data as well as the data for the selected env
+    api.request(`/issues/${group.id}/`, {
```

Comment:
Can we refactor these to use `api.requestPromise` while you're in there? (It's also fine to leave as-is)